src/compile/legend/assemble.ts

```diff
@@ -39,10 +39,20 @@ export function assembleLegends(model: Model): VgLegend[] {
     }
   }
 
-  return vals(legendByDomain)
+  const legends = vals(legendByDomain)
     .flat()
     .map(assembleLegend)
     .filter(l => l !== undefined);
+
+  if (model.config.aria === false) {
```

Comment:
In general, making side effect after a `map` should be avoided if possible.

Let's make `config` another parameter of `assembleLegend` instead.